flink-ml-benchmark/src/main/java/org/apache/flink/ml/benchmark/datagenerator/clustering/KMeansModelDataGenerator.java

```diff
@@ -55,20 +61,36 @@ public Table[] getData(StreamTableEnvironment tEnv) {
         InputDataGenerator<?> vectorArrayGenerator = new DenseVectorArrayGenerator();
         ReadWriteUtils.updateExistingParams(vectorArrayGenerator, paramMap);
         vectorArrayGenerator.setNumValues(1);
+        vectorArrayGenerator.setColNames(new String[] {"centroids"});
+
+        Table centroidsTable = vectorArrayGenerator.getData(tEnv)[0];
 
-        Table vectorArrayTable = vectorArrayGenerator.getData(tEnv)[0];
-        DataStream<KMeansModelData> modelDataStream =
-                tEnv.toDataStream(vectorArrayTable, DenseVector[].class)
-                        .map(new GenerateKMeansModelDataFunction());
+        Table modelDataTable =
+                centroidsTable.select(
+                        $("centroids"),
+                        call(GenerateWeightsFunction.class, $("centroids")).as("weights"));
```

Comment:
Can you explain why we need to make these changes and add names to those fields in the model data table?

Would it be better to keep the API (e.g. table format) consistent with that used by the production code (i.e. KMeansModel and KMeans)? For example, `KMeans#fit()` converts data stream to the model data table using `tEnv.fromDataStream(finalModelData)`.

Also, can we add a TODO/JIRA for creating nightly build to run all benchmarks nightly? We can use a small input number to verify that benchmark tools are broken.